subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,810 @@
+====
+
+Java projects typically include resources other than source files, such as properties files, that may need processing — for example by replacing tokens within the files — and packaging within the final JAR. The Java Plugin handles this by automatically creating a dedicated task for each defined source set called `process__SourceSet__Resources` (or `processResources` for the `main` source set). The following diagram shows how the source set fits in with this task:
+
+++++
+<figure>
+    <title>Processing non-source files for a source set</title>
+    <imageobject>
+        <imagedata fileref="img/java-sourcesets-process-resources.png" width="170mm"/>
+    </imageobject>
+</figure>
+++++
+
+As before, the shaded boxes represent properties of the source set, which in this case comprises the locations of the resource files and where they are copied to.
+
+In addition to the `main` source set, the Java Plugin defines a `test` source set that represents the project's tests. This source set is used by the `test` task, which runs the tests using https://junit.org/[JUnit] or http://testng.org/[TestNG].
+
+Projects typically use this source set for unit tests, but you can use it for integration, acceptance and other types of test if you wish. That said, most projects <<sec:custom_java_source_sets,define new source sets>> for those other test types because they require special setup or classpaths.
+
+You'll learn more about source sets and the features they provide in other sections of the chapter as many facets of building a Java project make use of them.
+
+[[sec:java_dependency_management_overview]]
+=== Managing your dependencies 
+
+The vast majority of Java projects rely on libraries, so managing a project's dependencies is an important part of building a Java project. Dependency management is a big topic, so we will focus on the basics for Java projects here. If you'd like to dive into the detail, check out the <<introduction_dependency_management,introduction to dependency management>>.
+
+Specifying the dependencies for your Java project requires just three pieces of information:
+
+ * Which dependency you need, such as a name and version
+ * What it's needed for, e.g. compilation or running
+ * Where to look for it
+
+The first two are specified in a `dependencies {}` block and the second in a `repositories {}` block. For example, to tell Gradle that your project requires version 3.6.7 of http://hibernate.org/[Hibernate] Core to compile and run your production code, and that you want to download the library from the Maven Central repository, you can use the following fragment:
+
+++++
+<sample id="javaPluginDependencyMgmt" dir="userguide/java/basic" title="Declaring dependencies">
+    <sourcefile file="build.gradle" snippet="java-dependency-mgmt"/>
+</sample>
+++++
+
+The Gradle terminology for the three elements is as follows:
```

Comment:
I think we should closely align the definition to https://docs.gradle.org/current/userguide/dependency_management_terminology.html.

Reply:
- [ ] Review and update

Reply:
Hmmm...that page doesn't mention dependency coordinates. Deliberate? Is there another, preferred term for them?

Reply:
Maybe just be an oversight. We have the module version in there which is just a part of the usual coordinates.

@gradle/dependency-management Does anything speak against adding "Module coordinates" as term there?

Reply:
I think we have the same confusion here as we had in other places. We should always use the term _module_ if we talk about the concrete thing we depend on. While the term _dependency_ should be used to describe the pointer at something.

So I would update the terminology here and to **module coordinates**:
> Module coordinates — the ID of the module to depend on, usually in the form...

And yes, I think we can (should?) add "Module coordinates" to dependency_management_terminology.html.